artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -120,6 +123,9 @@ public class ServerSessionImpl implements ServerSession, FailureListener {
 
    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
+   // TODO: how to configure this?
+   private int timeoutSessionClose = 5;
```

Comment:
@brusdev I haven't decided yet on where to have such configuration.. or if we could use another value in its place.

Reply:
System property perhaps?

That's TBD..


The test I wrote simulate the condition artificially

Reply:
it was too invasive on the API.. I am adding a configuration property on configuration, but I don't intend to add xml for it.. I will keep it at 30 seconds by default... ( a session should never take more than a few milliseconds to close )